crates/model/src/quote.rs

```diff
@@ -16,6 +15,20 @@ pub enum PriceQuality {
     Optimal,
 }
 
+#[derive(Eq, PartialEq, Clone, Copy, Debug, Default, Deserialize, Serialize, Hash)]
+#[serde(rename_all = "lowercase")]
+pub enum QuoteSigningScheme {
```

Comment:
An alternative implementation could look like this:
```
#[derive( Debug, Default, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub enum QuoteSigningScheme {
    #[default]
    Eip712,
    EthSign,
    Eip1271AsOnchainOrder,
    /// EIP orders can be off-chain and on-chain orders
    Eip1271 { 
    #[serde(default)]
    intended_onchain_order: bool
    },
    PreSign  { 
    #[serde(default)]
    intended_onchain_order: bool
    },
}
``

But I never managed to make it backwards compatible with the current API. Hence I decided for the current form. See my playground: https://play.rust-lang.org/?version=nightly&mode=release&edition=2021&gist=db2ec1559dc581fa97c01ad7e2a70ce3

Reply:
One thing I'm not super a fan of is introducing new `signingScheme` values that don't actually exist.

Here is a way that should be backwards compatible (basically it allows an additional `onchainOrder: bool` field in the JSON that gets "matched" up with the `signingScheme` field): https://play.rust-lang.org/?version=nightly&mode=release&edition=2021&gist=f32867ba6d117a26499b7ac8ee7939df

Reply:
Wow... the "tag" command is smart!

But unfortunately, it is not fully backwards compatible. Serde default and serde flatten do not work together, but your solution requires both:
https://github.com/serde-rs/serde/issues/1626

Here is your playground with a failing test: https://play.rust-lang.org/?version=nightly&mode=release&edition=2021&gist=68ed5b31ba40052762430b6d8e742002

I would love to take your solution, but I could not find a solution for that failing unit test

Reply:
> Serde default and serde flatten do not work together, but your solution requires both:

So close! Dang, I didn't know.

In this case, I would recommend deserializing to an intermediate structure and then
```rust
match (data.signing_scheme, data.onchain_order) {
            (scheme, true) if scheme.is_ecdsa_scheme() => bail!("ECDSA-signed orders cannot be on-chain"),
            // ...
        }
```

Something like: https://play.rust-lang.org/?version=nightly&mode=release&edition=2021&gist=bced89e0de1ccdfbe48e2960a424b3f7

What's since is that `serde` generates the code to convert from the intermediary struct to the `QuoteSigningScheme` struct without "leaking" the abstraction.